pkg/graveler/graveler.go

```diff
@@ -1089,19 +1101,40 @@ func (g *Graveler) GetRepositoryMetadata(ctx context.Context, repositoryID Repos
 	return g.RefManager.GetRepositoryMetadata(ctx, repositoryID)
 }
 
-func (g *Graveler) WriteRange(ctx context.Context, repository *RepositoryRecord, it ValueIterator) (*RangeInfo, error) {
+func (g *Graveler) WriteRange(ctx context.Context, repository *RepositoryRecord, it ValueIterator, opts ...SetOptionsFunc) (*RangeInfo, error) {
+	options := &SetOptions{}
+	for _, opt := range opts {
+		opt(options)
+	}
+	if !options.Force && repository.ReadOnly {
```

Comment:
relevant to all the checks - compare the repository.ReadOnly first, which is he common flow.